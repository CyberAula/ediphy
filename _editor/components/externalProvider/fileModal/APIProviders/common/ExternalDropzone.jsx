import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FileInput from '../../../../common/fileInput/FileInput';

/**
 * VISH Dropzone component
 */
export default class ExternalDropzone extends Component {

    state = {
        hover: false,
        file: null,
    };

    /**
     * Dropped file callback
     * @param acceptedFile
     * @param rejectedFile
     */
    onDrop = (event) => {
        let files = event.target.files;
        if (files.length === 1) {
            this.setState({ file: files[0] });
            if(this.props.callback) {
                this.props.callback(files[0]);
            }
        }
    };

    /**
     * Render React Component
     * @returns {code}
     */
    render() {
        return (
            <FileInput onChange={this.onDrop} className="fileInput" accept={this.props.accept}>
                <div className="fileDrag">
                    <span style={{ display: this.state.file ? 'none' : 'block' }}>
                        <b>{ Ediphy.i18n.t('FileInput.Drag') }</b>{ Ediphy.i18n.t('FileInput.Drag_2') }<b>{ Ediphy.i18n.t('FileInput.Click') }</b>{ Ediphy.i18n.t('FileInput.Click_2') }
                    </span>
                    <span className="fileUploaded" style={{ display: this.state.file ? 'block' : 'none' }}>
                        <i className="material-icons">insert_drive_file</i> { this.state.file ? this.state.file.name : '' }
                    </span>
                </div>
            </FileInput>
        );
    }

}

ExternalDropzone.propTypes = {
    /**
     * Mimetype accepted
     * */
    accept: PropTypes.any,
    /**
     * Function called when file is dropped
     */
    callback: PropTypes.func,
};
