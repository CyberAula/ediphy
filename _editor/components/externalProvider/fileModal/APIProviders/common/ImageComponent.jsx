import React from 'react';
import PropTypes from 'prop-types';
export default class ImageComponent extends React.Component {

    render() {
        let { isSelected, title, thumbnail, onElementSelected, url } = this.props;
        let avatar = thumbnail || url;
        return url ? <img onError={(e)=>{
            e.target.style.display = 'none';
        }} src={avatar}
        className={'catalogImage ' + (isSelected ? 'catalogImageSelected' : '')}
        title={title}
        onClick={() => {
            onElementSelected(title, url, 'image');
        }}
        /> : null;

    }
}
ImageComponent.propTypes = {
    /**
     * Boolean that indicates whether the user has selected the image
     */
    isSelected: PropTypes.bool,
    /**
     * Function that is called when the user selects the image
     */
    onElementSelected: PropTypes.func.isRequired,
    /**
     * Image thumbnail
     */
    thumbnail: PropTypes.string,
    /**
     * Image url
     */
    url: PropTypes.string,
    /**
     * Image title
     */
    title: PropTypes.string,
};
