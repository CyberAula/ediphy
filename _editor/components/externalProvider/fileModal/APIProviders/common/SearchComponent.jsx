import React from 'react';
import ReactDOM from 'react-dom';
import { FormControl, InputGroup, Glyphicon } from 'react-bootstrap';
import i18n from 'i18next';
import PropTypes from 'prop-types';

export default class SearchComponent extends React.Component {
    render() {
        const { disabled, onChange, onSearch } = this.props;
        return <InputGroup className="searchInputTopBar">
            <FormControl onChange={onChange} autoFocus ref="query" type="text" disabled={disabled}/>
            <InputGroup.Addon className="inputSearch" onClick={(e) => {
                onSearch(ReactDOM.findDOMNode(this.refs.query).value);
                e.preventDefault();
            }}>
                <Glyphicon glyph="search" />  {i18n.t("FileModal.APIProviders.search")}
            </InputGroup.Addon>
        </InputGroup>;
    }
}
SearchComponent.propTypes = {
    /**
     * Function called when the user clicks the search button
     */
    onSearch: PropTypes.func,
    /**
     * Function called when the user types on the search box
     */
    onChange: PropTypes.func,
    /**
   * Input disabled
   */
    disabled: PropTypes.bool,
};
