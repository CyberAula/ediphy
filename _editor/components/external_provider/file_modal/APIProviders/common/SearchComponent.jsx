import React from 'react';
import ReactDOM from 'react-dom';
import { Modal, FormControl, Col, Form, FormGroup, InputGroup, Glyphicon, ControlLabel, Button } from 'react-bootstrap';
import i18n from 'i18next';
import PropTypes from 'prop-types';

export default class SearchComponent extends React.Component {
    render() {
        return <InputGroup className="searchInputTopBar">
            <FormControl onChange={this.props.onChange} autoFocus ref="query" type="text" disabled={this.props.disabled}/>
            <InputGroup.Addon className="inputSearch" onClick={(e) => {
                this.props.onSearch(ReactDOM.findDOMNode(this.refs.query).value);
                e.preventDefault();
            }}>
                <Glyphicon glyph="search" />  {i18n.t("FileModal.APIProviders.search")}
            </InputGroup.Addon>
        </InputGroup>;
    }
}
SearchComponent.propTypes = {
    /**
     * Function called when the user clicks the search button
     */
    onSearch: PropTypes.func,
    /**
     * Function called when the user types on the search box
     */
    onChange: PropTypes.func,
    /**
   * Input disabled
   */
    disabled: PropTypes.bool,
};
