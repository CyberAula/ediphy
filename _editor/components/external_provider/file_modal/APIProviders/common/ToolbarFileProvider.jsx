import React, { Component } from 'react';
import { FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';
import i18n from 'i18next';
import PropTypes from 'prop-types';

export default class ToolbarFileProvider extends Component {
    /**
     * Constructor
     * @param props
     */
    constructor(props) {
        super(props);
        this.index = 0;
        /**
         * Component's initial state
         */
        this.state = {
            open: false,
        };
    }

    /**
     * Render React Component
     * @returns {XML}
     */
    render() {
        let bckg = this.props.formControlProps ? (this.props.value !== undefined ? this.props.value : this.props.formControlProps.value) : null;
        let isURI = (/data\:/).test(bckg);
        let props = { ...this.props.formControlProps,
            placeholder: isURI ? 'http://...' : '',
            value: isURI ? '' : bckg };
        return (<FormGroup style={{ display: this.props.hide ? "none" : "block" }}>
            {this.props.formControlProps ? [<ControlLabel key={1}>{props.label}</ControlLabel>,
                <FormControl key={2}{...props} onChange={e => {
                    props.onChange(e);
                }}/>] : null}
            <Button className={'toolbarButton'}
                onClick={() => {
                    this.setState({ open: true });
                    this.props.openModal(this.props.id, this.props.accept);
                }}>{this.props.buttontext}</Button>
        </FormGroup>);
    }
    componentWillReceiveProps(nextProps) {
        if(this.props.fileModalResult &&
            nextProps.fileModalResult &&
            nextProps.id === nextProps.fileModalResult.id
            && nextProps.fileModalResult.value &&
            this.state.open && this.props.fileModalResult.value !== nextProps.fileModalResult.value) {
            if (this.props.formControlProps) {
                this.props.formControlProps.onChange({ value: nextProps.fileModalResult.value });
            } else {
                this.props.onChange({ value: nextProps.fileModalResult.value });
            }
            this.setState({ open: false });
        }
    }
}

ToolbarFileProvider.propTypes = {
    /**
     * Toolbar input inherited props
     */
    formControlProps: PropTypes.object,
    /**
     * Input Value
     */
    value: PropTypes.any,
    /**
     * Open File Modal
     */
    openModal: PropTypes.func,
    /**
     * Id of the box/page being edited
     */
    id: PropTypes.any,
    /**
     * Mime Type accepted
     */
    accept: PropTypes.string,
    /**
     * Button text
     */
    buttontext: PropTypes.string,
    /**
     * Last files uploaded to server or searched in modal
     */
    fileModalResult: PropTypes.object,
    /**
     * Toolbar change callback
     */
    onChange: PropTypes.func,
    /**
     * Hide external provider input type
     */
    hide: PropTypes.bool,

};
