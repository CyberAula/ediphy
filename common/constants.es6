import i18n from 'i18next';
export const ID_PREFIX_SECTION = "se-";
export const ID_PREFIX_PAGE = "pa-";
export const ID_PREFIX_SLIDE = "sl-";
export const ID_PREFIX_BOX = "bo-";
export const ID_PREFIX_FILE = "file-";
export const ID_PREFIX_SORTABLE_BOX = "bs-";
export const ID_PREFIX_SORTABLE_CONTAINER = "sc-";
export const ID_PREFIX_CONTAINED_VIEW = "cv-";
export const ID_PREFIX_RICH_MARK = "rm-";
export const SECTIONS_HAVE_CONTENT = false;
export const FILE_UPLOAD_ERROR = 'FILE_UPLOAD_ERROR';
export const FILE_DELETE_ERROR = 'FILE_DELETE_ERROR';
export const FILE_UPLOADING = 'FILE_UPLOADING';
export const FILE_DELETING = 'FILE_DELETING';
export const FILE_NOT_ALLOWED = 'FILE_NOT_ALLOWED';
export const PAGE_TYPES = {
    SLIDE: "slide",
    DOCUMENT: "document",
    SECTION: "section",
};

export const extensions = [
    'all',
    'image',
    'audio',
    'video',
    'csv',
    'pdf',
    'scormpackage',
    'webapp',
    'swf',
    'xml',
    'obj',
    'edi',
    'vish',
];

export const MESSAGE_TYPES = {
    NOTIFICATION: "notification",
    FAVOR: "favor",
};

MESSAGE_TYPES.NOTIFICATION;
