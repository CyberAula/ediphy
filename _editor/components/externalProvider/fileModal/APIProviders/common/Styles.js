import styled from 'styled-components';
import { PRIMARY_BLUE } from "../../../../../../sass/general/constants";

export const ImageCatalog = styled.img`
    width: 100%;
    height: auto;
    min-height:180px;
    max-height:200px;
    border: solid ${props => props.selected ? PRIMARY_BLUE : 'transparent'} 3px;
`;
