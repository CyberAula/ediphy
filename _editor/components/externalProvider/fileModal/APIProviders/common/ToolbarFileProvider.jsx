import React, { Component } from 'react';
import { FormGroup, FormControl, ControlLabel } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { ToolbarButton } from "../../../../toolbar/toolbarComponents/Styles";

export default class ToolbarFileProvider extends Component {

    /**
     * Render React Component
     * @returns {XML}
     */
    render() {
        const { accept, buttontext, formControlProps, hide, id, value, onChange, openModal } = this.props;
        const bckg = value ?? formControlProps?.value ?? null;
        const isURI = (/data\:/).test(bckg);
        const props = { ...formControlProps, placeholder: isURI ? 'http://...' : '', value: isURI ? '' : bckg };
        return (
            <FormGroup style={{ display: hide ? "none" : "block" }}>
                {formControlProps ? [<ControlLabel key={1}>{props.label}</ControlLabel>,
                    <FormControl key={2}{...props} onChange={e => {
                        onChange({ value: e.target.value });
                    }}/>] : null}
                <ToolbarButton
                    onClick={() => {
                        openModal(id, accept);
                    }}>{buttontext}</ToolbarButton>
            </FormGroup>);
    }
    UNSAFE_componentWillReceiveProps(nextProps) {
        if(nextProps.id === nextProps.fileModalResult?.id
            && ((!this.props.fileModalResult && nextProps.fileModalResult.value)
            || (this.props.fileModalResult.value !== nextProps.fileModalResult.value))) {
            this.props.onChange({ value: nextProps.fileModalResult.value });
        }
    }
}

ToolbarFileProvider.propTypes = {
    /**
     * Toolbar input inherited props
     */
    formControlProps: PropTypes.object,
    /**
     * Input Value
     */
    value: PropTypes.any,
    /**
     * Open File Modal
     */
    openModal: PropTypes.func,
    /**
     * Id of the box/page being edited
     */
    id: PropTypes.any,
    /**
     * Mime Type accepted
     */
    accept: PropTypes.string,
    /**
     * Button text
     */
    buttontext: PropTypes.string,
    /**
     * Last files uploaded to server or searched in modal
     */
    fileModalResult: PropTypes.object,
    /**
     * Toolbar change callback
     */
    onChange: PropTypes.func,
    /**
     * Hide external provider input type
     */
    hide: PropTypes.bool,

};
