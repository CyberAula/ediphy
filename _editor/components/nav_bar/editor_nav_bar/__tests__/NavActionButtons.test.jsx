import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ID_PREFIX_BOX, ID_PREFIX_SECTION } from '../../../../../common/constants';

let NavActionButtons;

// Dependencies' mocks
jest.mock('i18next', () => ({
    t: () => 'i18n string',
}));

beforeEach(() => {
    jest.mock('screenfull', () => ({
        on: jest.fn().mockName('on'),
        off: jest.fn().mockName('off'),
        toggle: jest.fn().mockName('toggle'),
    }));

    // Ediphy.Config is set to make the snapshot deterministic, regardless of
    // the actual config in the repo
    jest.mock('../../../../../core/editor/main', () => ({
        Config: {
            disable_save_button: false,
            publish_button: false,
        },
    }));

    // Reset all modules with the new mocks and reload the component
    jest.resetModules();
    NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

    // Forget about any mock function being called previously
    jest.clearAllMocks();
});

// Unit test's mocks
const mockFunctions = {
    changeGlobalConfig: jest.fn().mockName('changeGlobalConfig'),
    onTextEditorToggled: jest.fn().mockName('onTextEditorToggled'),
    redo: jest.fn().mockName('redo'),
    save: jest.fn().mockName('save'),
    serverModalOpen: jest.fn().mockName('serverModalOpen'),
    undo: jest.fn().mockName('undo'),
    visor: jest.fn().mockName('visor'),
};

const placeholderProps = {
    boxSelected: 0,
    globalConfig: {},
    navItems: {
        'MOCK_NAV_ITEM': {},
    },
    navItemSelected: 'MOCK_NAV_ITEM',
    redoDisabled: false,
    undoDisabled: false,
    ...mockFunctions,
};

describe('NavActionButtons - General', () => {
    it('should render with required props', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it('should set the screenfull onChange handler after mounting the component', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        const screenfull = jest.requireMock('screenfull');
        expect(screenfull.on).toBeCalledWith('change', component.instance().checkFullScreen);
    });

    it('should remove the screenfull onChange handler before unmounting the component', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Trigger the unmount lifecycle to invoke the componentWillUnmount() method
        component.unmount();

        const screenfull = jest.requireMock('screenfull');
        expect(screenfull.off).toBeCalled();
    });

    it('should save in state.isFullScreenOn the full screen status when checkFullScreen() is called', () => {
        // Override screenfull's mock
        jest.mock('screenfull', () => ({
            on: jest.fn().mockName('on'),
            off: jest.fn().mockName('off'),
            isFullscreen: true,
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Call the component's checkFullScreen method
        component.instance().checkFullScreen();

        expect(component.state('isFullScreenOn')).toBeTruthy();
    });
});

describe('NavActionButtons - "Fullscreen" button', () => {
    it('should have the "fullscreen" icon when state.isFullScreenOn = false', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Explicitly set state.isFullScreenOn = false
        component.setState({ isFullScreenOn: false });
        expect(component.find('button[name="fullscreen"] i').first().text()).toEqual('fullscreen');
    });

    it('should have the "fullscreen_exit" icon when state.isFullScreenOn = true', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Set state.isFullScreenOn = true
        component.setState({ isFullScreenOn: true });
        expect(component.find('button[name="fullscreen"] i').first().text()).toEqual('fullscreen_exit');
    });

    it('should toggle the full screen when the button is clicked', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Click the "Fullscreen" button
        component.find('button[name="fullscreen"]').first().simulate('click');

        const screenfull = jest.requireMock('screenfull');
        expect(screenfull.toggle).toBeCalled();
    });
});

describe('NavActionButtons - "Undo" button', () => {
    it('should disable the button when undoDisabled = true', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} undoDisabled />);

        // "Undo" button
        expect(component.find('button[name="undo"]').first().prop('disabled')).toBeTruthy();
    });
});

describe('NavActionButtons - "Redo" button', () => {
    it('should disable the button when redoDisabled = true', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} redoDisabled />);

        // "Redo" button
        expect(component.find('button[name="redo"]').first().prop('disabled')).toBeTruthy();
    });
});

describe('NavActionButtons - "Save" button', () => {
    it('should display the button when configured and publish_button = undefined', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                disable_save_button: false,
                publish_button: undefined,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps} />);

        expect(component.find('button[name="save"]')).toHaveLength(1);
    });

    it('should display the button when configured and publish_button = false', () => {
        // Override Ediphy.Config's mock (again, this time with publish_button set up)
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                disable_save_button: false,
                publish_button: false,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps} />);

        expect(component.find('button[name="save"]')).toHaveLength(1);
    });

    it('should handle when the button is clicked', () => {
        const component = shallow(<NavActionButtons {...placeholderProps} />);

        // Click the "Save" button
        component.find('button[name="save"]').first().simulate('click');

        expect(mockFunctions.save).toBeCalled();
        expect(mockFunctions.serverModalOpen).toBeCalled();
    });
});

describe('NavActionButtons - "Publish" button', () => {
    it('should display the button when configured and the course has "draft" status', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'draft' }} />);

        expect(component.find('button[name="publish"]')).toHaveLength(1);
    });

    it('should handle when the button is clicked', () => {
        // Here we force again the creation of the button, in order to be
        // able to run rests on it

        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'draft' }} />);

        // Click the "Save" button
        component.find('button[name="publish"]').first().simulate('click');

        expect(mockFunctions.changeGlobalConfig).toBeCalledWith('status', 'final');
        expect(mockFunctions.save).toBeCalled();
        expect(mockFunctions.serverModalOpen).toBeCalled();
    });
});

describe('NavActionButtons - "Unpublish" button', () => {
    it('should display the button when configured and the course has "final" status', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'final' }} />);

        expect(component.find('button[name="unpublish"]')).toHaveLength(1);
    });

    it('should handle when the button is clicked', () => {
        // Here we force again the creation of the button, in order to be
        // able to run rests on it

        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'final' }} />);

        // Click the "Save" button
        component.find('button[name="unpublish"]').first().simulate('click');

        expect(mockFunctions.changeGlobalConfig).toBeCalledWith('status', 'draft');
        expect(mockFunctions.save).toBeCalled();
        expect(mockFunctions.serverModalOpen).toBeCalled();
    });
});

describe('NavActionButtons - "Preview" button', () => {
    it('should disable the button when navItemSelected = 0', () => {
        const component = shallow(<NavActionButtons {...placeholderProps}
            navItemSelected={0} />);

        expect(component.find('button[name="preview"]').first().prop('disabled')).toBeTruthy();
    });

    it('should disable the button when a section is selected and sections can\'t have content', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                sections_have_content: false,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            navItemSelected={`${ID_PREFIX_SECTION}-3141592653590`} />);

        expect(component.find('button[name="preview"]').first().prop('disabled')).toBeTruthy();
    });

    it('should handle when the button is clicked (with a box selected)', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const mockedBox = `${ID_PREFIX_BOX}-3141592653590`;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'final' }}
            boxSelected={ mockedBox } />);

        // Click the "Save" button
        component.find('button[name="preview"]').first().simulate('click');

        expect(mockFunctions.onTextEditorToggled).toBeCalledWith(mockedBox, false);
        expect(mockFunctions.visor).toBeCalled();
    });

    it('should handle when the button is clicked (without any box selected)', () => {
        // Override Ediphy.Config's mock
        jest.mock('../../../../../core/editor/main', () => ({
            Config: {
                publish_button: true,
            },
        }));
        jest.resetModules();
        NavActionButtons = jest.requireActual('../NavActionButtons.jsx').default;

        const component = shallow(<NavActionButtons {...placeholderProps}
            globalConfig={{ status: 'final' }}
            boxSelected={0} />);

        // Click the "Save" button
        component.find('button[name="preview"]').first().simulate('click');

        expect(mockFunctions.onTextEditorToggled).not.toBeCalled();
        expect(mockFunctions.visor).toBeCalled();
    });
});
