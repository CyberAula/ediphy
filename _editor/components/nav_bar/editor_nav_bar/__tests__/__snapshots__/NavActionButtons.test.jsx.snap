// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavActionButtons - General should render with required props 1`] = `
<div
  className="navButtons"
>
  <button
    className="navButton navbarButton_fullscreen"
    disabled={false}
    key="fullscreen"
    name="fullscreen"
    onClick={[Function]}
    title="i18n string"
  >
    <i
      className="material-icons"
    >
      fullscreen
    </i>
    <br />
    <span
      className="hideonresize"
    >
      i18n string
    </span>
  </button>
  <button
    className="navButton navbarButton_undo"
    disabled={false}
    key="undo"
    name="undo"
    onClick={[MockFunction undo]}
    title="i18n string"
  >
    <i
      className="material-icons"
    >
      undo
    </i>
    <br />
    <span
      className="hideonresize"
    >
      i18n string
    </span>
  </button>
  <button
    className="navButton navbarButton_redo"
    disabled={false}
    key="redo"
    name="redo"
    onClick={[MockFunction redo]}
    title="i18n string"
  >
    <i
      className="material-icons"
    >
      redo
    </i>
    <br />
    <span
      className="hideonresize"
    >
      i18n string
    </span>
  </button>
  <button
    className="navButton navbarButton_save"
    disabled={false}
    key="save"
    name="save"
    onClick={[Function]}
    title="i18n string"
  >
    <i
      className="material-icons"
    >
      save
    </i>
    <br />
    <span
      className="hideonresize"
    >
      i18n string
    </span>
  </button>
  <button
    className="navButton navbarButton_preview"
    disabled={false}
    key="preview"
    name="preview"
    onClick={[Function]}
    title="i18n string"
  >
    <i
      className="material-icons"
    >
      visibility
    </i>
    <br />
    <span
      className="hideonresize"
    >
      i18n string
    </span>
  </button>
</div>
`;
